### ISSUE TITLE
Unable to access jarfile $JDTLS_HOME/plugins/org.eclipse.equinox.launcher_*.jar

### ISSUE THREAD

**Issue by trevorwhitney** [mfussenegger/nvim-jdtls#242]
### LSP client configuration

```lua
function Java.configure_lsp(on_attach, capabilities)
	local project_name = vim.fn.fnamemodify(vim.fn.getcwd(), ":p:h:t")
        local home = vim.loop.os_getenv("HOME")

	-- See `:help vim.lsp.start_client` for an overview of the supported `config` options.
	local config = {
		-- The command that starts the language server
		-- See: https://github.com/eclipse/eclipse.jdt.ls#running-from-the-command-line
		cmd = {

			-- 💀
			"java", -- or '/path/to/java11_or_newer/bin/java'
			-- depends on if `java` is in your $PATH env variable and if it points to the right version.

			"-Declipse.application=org.eclipse.jdt.ls.core.id1",
			"-Dosgi.bundles.defaultStartLevel=4",
			"-Declipse.product=org.eclipse.jdt.ls.core.product",
			"-Dlog.protocol=true",
			"-Dlog.level=ALL",
			"-Xms1g",
			"--add-modules=ALL-SYSTEM",
			"--add-opens",
			"java.base/java.util=ALL-UNNAMED",
			"--add-opens",
			"java.base/java.lang=ALL-UNNAMED",

			-- 💀
			"-jar",
			home .. "/.local/opt/jdtls/plugins/org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar",
			-- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                       ^^^^^^^^^^^^^^
			-- Must point to the                                                     Change this to
			-- eclipse.jdt.ls installation                                           the actual version

			-- 💀
			"-configuration",
			home .. "/.local/opt/jdtls/config_linux",
			-- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^        ^^^^^^
			-- Must point to the                      Change to one of `linux`, `win` or `mac`
			-- eclipse.jdt.ls installation            Depending on your system.

			-- 💀
			-- See `data directory configuration` section in the README
			"-data",
			home .. "/.local/opt/jdtls/workspace/" .. project_name,
		},

		-- 💀
		-- This is the default if not provided, you can remove it. Or adjust as needed.
		-- One dedicated LSP server & client will be started per unique root_dir
		root_dir = require("jdtls.setup").find_root({ ".git", "mvnw", "gradlew", "build.gradle" }),

		-- Here you can configure eclipse.jdt.ls specific settings
		-- See https://github.com/eclipse/eclipse.jdt.ls/wiki/Running-the-JAVA-LS-server-from-the-command-line#initialize-request
		-- for a list of options
		settings = {
			java = {},
		},

		-- Language server `initializationOptions`
		-- You need to extend the `bundles` with paths to jar files
		-- if you want to use additional eclipse.jdt.ls plugins.
		--
		-- See https://github.com/mfussenegger/nvim-jdtls#java-debug-installation
		--
		-- If you don't plan on using the debugger or other eclipse.jdt.ls plugins you can remove this
		init_options = {
			bundles = {},
		},

    on_attach = function(client, bufnr)
      require("jdtls").add_commands()
      on_attach(client, bufnr)
    end
	}
	-- This starts a new client & server,
	-- or attaches to an existing client & server depending on the `root_dir`.
	require("jdtls").start_or_attach(config)
end
```

### Eclipse.jdt.ls version

1.6.400.v20210924-0641

### Steps to Reproduce

In a project using gradle 7.2 with gradle wrapper and openjdk 11

1. open neovim in the root of the project
2. see message that jdtls is starting, see `ServiceReady` message
3. open a java file
4. see `Client 3 quit with exit code 1 and signal 0`
5. In the lsp log, see `[ERROR][2022-03-01 14:47:28] .../vim/lsp/rpc.lua:420    "rpc"   "/nix/store/6l9r21ic68mavp8cqbv6k774j0hv7mjg-openjdk-11.0.12+7/lib/openjdk/bin/java"    "stderr"        "Error: Unable to access jarfile $JDTLS_HOME/plugins/org.eclipse.equinox.launcher_*.jar\n"`

### Expected Result

Opening the java file would not cause the lsp client to crash.

### Actual Result

lsp client crashes with error: `ERROR][2022-03-01 14:47:28] .../vim/lsp/rpc.lua:420    "rpc"   "/nix/store/6l9r21ic68mavp8cqbv6k774j0hv7mjg-openjdk-11.0.12+7/lib/openjdk/bin/java"    "stderr"        "Error: Unable to access jarfile $JDTLS_HOME/plugins/org.eclipse.equinox.launcher_*.jar\n"`

**Comment by trevorwhitney** [comment]
I was testing with a canned project if that helps: https://github.com/cloudfoundry-samples/spring-music

**Comment by trevorwhitney** [comment]
It seems to work if I just export JDTLS_HOME to the correct location. I'll add that to my environment, but I didn't see this mentioned in any of the jdtls installation docs, did I just miss it?

**Comment by EpsilonKu** [comment]
Are you sure that `.local/opt/jdtls/` is not read-only? I suggest to put it on `.local/bin` or on your home directory.

**Comment by mfussenegger** [comment]
The error message you posted doesn't seem to match the configuration you posted. Are you sure you're not having some additional lspconfig stuff active or are pulling in something else via some installer/framework stuff?

**Comment by trevorwhitney** [comment]
So I ran into a few problems:
1) I did have folder permission issues in the location jdtls was extracted to, didn't realize that had to be writable by the process running jdtls. That's been fixed/
2) I do still need to export "JDTLS_HOME", which I'm now doing in my `init.vim`

With these fixes everything seems to be working now for code in the project, however I can't go to definition for any library code (ie. spring source code).

That is the config I'm using (except that I moved jdtls to `~/.local/share/jdtls` since posting). @mfussenegger I am using lspconfig, but am not calling `setup` for `jdtls1 (ie. `require'lspconfig'.jdtls.setup{}`). However it does look like that `JDTLS_HOME` path is coming from lspconfig. Is there something else I need to explicitly disable that?
